DentOS_Framework/DentOsTestbedLib/src/dent_os_testbed/lib/lldp/linux/linux_lldp_impl.py

```diff
@@ -22,24 +20,46 @@ def format_show(self, command, *argv, **kwarg):
         params = kwarg['params']
         if params.get('dut_discovery', False):
             params['cmd_options'] = '-f json'
-        cmd = 'lldpctl {} '.format(params.get('cmd_options', ''))
+        if 'lldpctl' in kwarg['params']:
+            cmd = 'lldpctl{} {} '.format(params.get('cmd_options', ''), command)
+        else:
+            cmd = 'lldpcli{} {} '.format(params.get('cmd_options', ''), command)
+        if 'neighbors' in params and params['neighbors']:
+            cmd += 'neighbors '
+        if 'statistics' in params and params['statistics']:
+            cmd += 'statistics '
+        if 'ports' in params and params['ports']:
+            cmd += 'ports'
         if 'interface' in kwarg['params']:
             cmd += ' {} '.format(kwarg['params']['interface'])
         return cmd
 
     def parse_show(self, command, output, *argv, **kwarg):
-        interfaces = []
-        try:
-            parsed_out = json.loads(output)
-            for interface in parsed_out['lldp']['interface']:
-                for port, data in interface.items():
-                    item = {'interface': port, 'remote_interface': data['port']['id']['value']}
-                    for chassis in data['chassis']:
-                        item['remote_host'] = chassis
-                    interfaces.append(item)
-        except Exception:
-            return []
-        return interfaces
+        lines = output.strip().split('\n')
+        lines = lines[2:-2]
+        lldp_dict = {}
+        for line in lines:
+            key_value_pair = line.split(':')
+            if len(key_value_pair) != 2:
+                continue
+            key = key_value_pair[0].strip()
+            value = key_value_pair[1].strip()
+            lldp_dict[key] = value
+        return (lldp_dict)
+
+    # def parse_show(self, command, output, *argv, **kwarg):
+    #     interfaces = []
+    #     try:
+    #         parsed_out = json.loads(output)
+    #         for interface in parsed_out['lldp']['interface']:
+    #             for port, data in interface.items():
+    #                 item = {'interface': port, 'remote_interface': data['port']['id']['value']}
+    #                 for chassis in data['chassis']:
+    #                     item['remote_host'] = chassis
+    #                 interfaces.append(item)
+    #     except Exception:
+    #         return []
+    #     return interfaces
```

Comment:
Instead of commenting out the code, you can catch JSONDecodeError which will occur if the output is not a JSON-format and there go with your parsing algorithm